tests/v2/validation/fleet/provisioning/airgap/fleet_in_airgap_test.go

```diff
@@ -0,0 +1,214 @@
+	provisioning "github.com/rancher/rancher/tests/v2/actions/provisioning"
+	"github.com/rancher/rancher/tests/v2/actions/provisioning/permutations"
+	"github.com/rancher/rancher/tests/v2/actions/provisioninginput"
+	"github.com/rancher/rancher/tests/v2/actions/reports"
+	"github.com/rancher/shepherd/clients/corral"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	extensionsfleet "github.com/rancher/shepherd/extensions/fleet"
+	"github.com/rancher/shepherd/extensions/users"
+	password "github.com/rancher/shepherd/extensions/users/passwordgenerator"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AirGapRKE2CustomClusterTestSuite struct {
+	suite.Suite
+	client         *rancher.Client
+	standardClient *rancher.Client
+	session        *session.Session
+	corralPackage  *corral.Packages
+	clustersConfig *provisioninginput.Config
+	registryFQDN   string
+	fleetGitRepo   *v1alpha1.GitRepo
+}
+
+func (a *AirGapRKE2CustomClusterTestSuite) TearDownSuite() {
+	a.session.Cleanup()
+}
+
+func (a *AirGapRKE2CustomClusterTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	a.session = testSession
+
+	a.clustersConfig = new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, a.clustersConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(a.T(), err)
+
+	a.client = client
+
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	enabled := true
+
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(a.T(), err)
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(a.T(), err)
+
+	a.standardClient = standardUserClient
+
+	corralConfig := corral.Configurations()
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	a.corralPackage = corral.PackagesConfig()
+
+	bastionIP := corralConfig.CorralConfigVars[corralBastionIP]
+	bastionUser := corralConfig.CorralConfigVars[corralSSHUser]
+
+	// format the privateKey string for use with ssh package.
+	privateKey := corralConfig.CorralConfigVars[corralPrivateKey]
+	privateKey = strings.Replace(privateKey, "\\n", "\n", -1)
+	privateKey = strings.Replace(privateKey, "\"", "", -1)
```

Comment:
```suggestion
privateKey := corralConfig.CorralConfigVars[corralPrivateKey]
privateKey = strings.NewReplacer("\\n", "\n", "\"", "").Replace(privateKey)
```

Reference: https://pkg.go.dev/strings#NewReplacer

Reply:
can this be in a separate PR, we have use strings.Replace in provisioning too. would make sense to update it all at once.